src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -148,6 +146,8 @@ public ProNonCombatMoveAI(final ProAI ai) {
           break;
         }
       }
+    } else {
+      moveUnitsToBestTerritories();
```

Comment:
Not following you here. You need the while loop since the code can run multiple times before running into the break condition. The idea is it prioritizes territories near its capital more and more each iteration if the so is more conservative. This is to make sure the AI doesn't move too many units away from its capital and a enemy stack can march straight for it.